datamade/la-metro-dashboard
# Create production deployment

## Description

This PR creates a production deployment for the Metro dashboard. It has replaced the staging dashboard at https://la-metro-dashboard.datamade.us. Meanwhile, staging has been moved to https://la-metro-dashboard-staging.datamade.us.

Closes #42.

## Notes

Related changes in external repos / outside of version control:

- Spun up `LA Metro Dashboard – Production` EC2 instance, added to `LA Metro Dashboard` security group allowing outbound connections on 5432.
- Updated `la-metro-councilmatic` security group to allow inbound connections to Postgres and Solr from the production IP.
- Updated Metro server `pg_hba.conf` to accept connections from the production IP: https://github.com/datamade/server-la-metro-councilmatic-upgrade/commit/c2642e1e82a9b2e85b70fd8d9c7963bb20e359df
- Updated the production Solr container in LA Metro Councilmatic to run on `0.0.0.0`: https://github.com/datamade/la-metro-councilmatic/pull/644
- Shelled into the production dashboard server, changed into the application directory, activated the virtual environment, and created a user account for myself using [the `create_user` airflow CLI command](https://airflow.apache.org/docs/stable/cli-ref#create_user): `AIRFLOW_HOME=$(pwd) airflow create_user ...`
- Created a datamade user account via the Airflow UI and added credentials to the team LastPass.

## Testing instructions

- I've manually triggered both staging and production deployments from the latest commit. If you want to reproduce this:
  - Copy `46689e1914e7c5765e81f9e6f00085085d6a7e2a` (the latest commit at the time of writing).
  - Copy the latest deployment for staging in CodeDeploy, paste in the commit hash, and trigger the new deployment. Confirm it succeeds. View the staging site and confirm it looks good.
  - Repeat these steps for the production deployment group.
- I have also run the `windowed_bill_scraping` and `hourly_processing` DAGs to confirm they can access the production database and Solr index. If you wish to reconfirm, turn on the DAG and, if a run does not start automatically, trigger a run and confirm it succeeds.
- Confirm you can log into the production dashboard with the team login associated with `metro-support@datamade.us` in our shared LastPass folder: https://la-metro-dashboard.datamade.us/login/
